### PR TITLE
Update go version to 1.18

### PR DESCRIPTION
**Pull request by arihantg**

### CODE REVIEW

File: .github/workflows/go.yml [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+
+jobs:
+  golangci-lint:
+    name: Golangci-lint
+    strategy:
+      matrix:
+        platform: [ ubuntu-18.04 ]
+    runs-on: ${{ matrix.platform }}
+    steps:
+    - name: Set up Go 1.15
```
Comment (swathinsankaran) [inline .github/workflows/go.yml:21]: Do we have to upgrade the version to 1.18 here?

File: .github/workflows/go.yml [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+      uses: actions/setup-go@v1
+      with:
+        go-version: 1.15
```
Comment (swathinsankaran) [inline .github/workflows/go.yml:24]: We have to upgrade the version to 1.18 here.

File: gslb/test/integration/custom_fqdn/hostrule_test.go [hunk 3]
```diff
@@ -603,10 +607,12 @@ func TestHRCreateUpdateDuplicateAliasesInCluster(t *testing.T) {
 	newK8sHr := getTestHostRule(t, K8s, k8sHr.Name, k8sHr.Namespace)
 	newK8sHr.Spec.VirtualHost.Aliases = []string{"dupK8s_alias.avi.com", "dupK8s_alias.avi.com"}
 	newK8sHr.Status.Status = gslbutils.HostRuleRejected
-	updateHostRule(t, K8s, newK8sHr)
+	// updateHostRule(t, K8s, newK8sHr)
+	updateHostRuleStatus(t, K8s, newK8sHr)
 
 	// ingMember is removed from expectedMembers as the hostrule is rejected
 	expectedMembers = []nodes.AviGSK8sObj{getTestGSMemberFromRoute(t, routeObj, routeCluster, 1)}
+	t.Logf("SWATHIN getTestGSMemberFromRoute %+v", expectedMembers)
```
Comment (swathinsankaran) [inline gslb/test/integration/custom_fqdn/hostrule_test.go:615]: we can remove this log.

File: gslb/test/integration/custom_fqdn/hostrule_test.go [hunk 4]
```diff
@@ -584,7 +584,10 @@ func TestHRCreateUpdateDuplicateAliasesInCluster(t *testing.T) {
 		gslbutils.HostRuleAccepted, nil, true)
 	createHostRule(t, K8s, k8sHr)
 
+	t.Logf("SWATHIN expectedMembers %+v", expectedMembers)
+
 	expectedMembers = append(expectedMembers, getTestGSMemberFromIng(t, ingObj, ingCluster, 1))
+	t.Logf("SWATHIN expectedMembers %+v", expectedMembers)
```
Comment (swathinsankaran) [inline gslb/test/integration/custom_fqdn/hostrule_test.go:590]: same here

File: gslb/test/integration/custom_fqdn/hostrule_test.go [hunk 5]
```diff
@@ -595,6 +598,7 @@ func TestHRCreateUpdateDuplicateAliasesInCluster(t *testing.T) {
 	createHostRule(t, Oshift, ocHr)
 
 	expectedMembers = append(expectedMembers, getTestGSMemberFromRoute(t, routeObj, routeCluster, 1))
+	t.Logf("SWATHIN getTestGSMemberFromRoute %+v", expectedMembers)
```
Comment (swathinsankaran) [inline gslb/test/integration/custom_fqdn/hostrule_test.go:601]: same here

File: gslb/test/integration/custom_fqdn/hostrule_test.go [hunk 6]
```diff
@@ -603,10 +607,12 @@ func TestHRCreateUpdateDuplicateAliasesInCluster(t *testing.T) {
 	newK8sHr := getTestHostRule(t, K8s, k8sHr.Name, k8sHr.Namespace)
 	newK8sHr.Spec.VirtualHost.Aliases = []string{"dupK8s_alias.avi.com", "dupK8s_alias.avi.com"}
 	newK8sHr.Status.Status = gslbutils.HostRuleRejected
-	updateHostRule(t, K8s, newK8sHr)
+	// updateHostRule(t, K8s, newK8sHr)
```
Comment (swathinsankaran) [inline gslb/test/integration/custom_fqdn/hostrule_test.go:610]: we can remove this commented code.

File: gslb/test/integration/custom_fqdn/hostrule_test.go [hunk 7]
```diff
@@ -615,7 +621,8 @@ func TestHRCreateUpdateDuplicateAliasesInCluster(t *testing.T) {
 	newOcHr := getTestHostRule(t, Oshift, ocHr.Name, ocHr.Namespace)
 	newOcHr.Spec.VirtualHost.Aliases = []string{"dupOc_alias.avi.com", "dupOc_alias.avi.com"}
 	newOcHr.Status.Status = gslbutils.HostRuleRejected
-	updateHostRule(t, Oshift, newOcHr)
+	// updateHostRule(t, Oshift, newOcHr)
```
Comment (swathinsankaran) [inline gslb/test/integration/custom_fqdn/hostrule_test.go:624]: this commented code is not required. Can you check other places also?

**Review by swathinsankaran — APPROVED** [review]

**Comment by swathinsankaran** [comment]
build amko

File: Makefile [hunk 8]
```diff
@@ -7,10 +7,18 @@ GOTEST=$(GOCMD) test
...
 
+
+ifdef GOLANG_SRC_REPO
+	BUILD_GO_IMG=$(GOLANG_SRC_REPO)
+else
+	BUILD_GO_IMG=golang:latest
+endif
+
 .PHONY: all
 all: vendor build
```
Comment (swathinsankaran) [inline Makefile:24]: build changes are missing. please refer this PR [https://github.com/vmware/global-load-balancing-services-for-kubernetes/pull/176/files#diff-76ed074a9305c04054cdebb9[…]07091de1f20cad0bbac34ffb52R26](https://github.com/vmware/global-load-balancing-services-for-kubernetes/pull/176/files#diff-76ed074a9305c04054cdebb9e9aad2d818052b07091de1f20cad0bbac34ffb52R26)

Comment (arihantg) [reply Makefile:24]: this is taken care now.

**Pull request merged**